src/code_8002774.c

```diff
@@ -66,3 +154,342 @@ s32 sub_8002B04(s32 _direction1, s32 _direction2)
     }
     return direction1;
 }
+
+UNUSED s32 sub_8002B5C(s32 _direction1, s32 _direction2)
+{
+    int newDirection;
+    int direction1;
+    int direction2;
+
+    direction1 = (s8)_direction1;
+    direction2 = (s8)_direction2;
+
+    if (direction2 != -1) {
+        newDirection = ((direction2 + NUM_DIRECTIONS) - direction1) % NUM_DIRECTIONS;
+
+        if (newDirection != 0) {
+            if (newDirection < 4) {
+                newDirection = (direction1 + 2) % NUM_DIRECTIONS;
+            }
+            else {
+                newDirection = (direction1 + 6) % NUM_DIRECTIONS;
+            }
+            direction1 = (s8)(newDirection);
+        }
+    }
+    return direction1 & DIRECTION_MASK_CARDINAL;
+}
+
+NAKED void SetVecFromDirectionSpeed(void)
```

Comment:
Matched https://decomp.me/scratch/ErmNR

Reply:
Thanks, added it as a commit in the branch.